handle/handle.go

```diff
@@ -152,6 +153,45 @@ func AddCorsWildcardHeaders(serve http.HandlerFunc) http.HandlerFunc {
 	}
 }
 
+// Access Control through url parameters. The access key is set by ACCESS_KEY.
+// md5sum is computed by queried path + access key
+// (e.g. "/my/file" + ACCESS_KEY)
+func AddAccessKey(serve http.HandlerFunc, accessKey string) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		// Get key or md5sum from this access.
+		keys, key_ok := r.URL.Query()["key"]
+		var code string
+    	if !key_ok || len(keys[0]) < 1 {
+			// In case a code is provided
+			codes, code_ok := r.URL.Query()["code"]
+    		if !code_ok || len(codes[0]) < 1 {
+				http.NotFound(w, r)
+				return
+			}
+			code = codes[0]
+		} else {
+			// In case a key is provided, convert to code.
+			data := []byte(r.URL.Path + keys[0])
+			hash := md5.Sum(data)
+			code = fmt.Sprintf("%x", hash)
```

Comment:
```go
code = fmt.Sprintf("x", hash)
```
->
```go
code = fmt.Sprintf("%X", hash)
```
This will make the hexadecimal output upper-case. If you make this change, you can move the `code = strings.ToUpper(code)` into the `if` clause as a performance improvement.